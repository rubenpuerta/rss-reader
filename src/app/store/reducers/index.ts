export * from './app.reducer';
export * from './ui.reducer';
