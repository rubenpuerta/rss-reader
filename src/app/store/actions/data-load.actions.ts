import { createAction, props, union } from '@ngrx/store';

import { NewsItem } from '@models/rss-news.model';

const initialDataLoad = createAction('[RSS] Initial data load');
const initialDataLoadSuccess = createAction(
	'[RSS] Initial data load success',
	props<{ news: NewsItem[] }>()
);
const initialDataLoadFailure = createAction('[RSS] Initial data load failure');

const actions = { initialDataLoad, initialDataLoadSuccess, initialDataLoadFailure };

const actionsUnions = union(actions);

export type DataLoadctionsType = typeof actionsUnions;
export const DataLoadActions = actions;
