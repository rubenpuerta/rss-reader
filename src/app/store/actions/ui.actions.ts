import { createAction, union } from '@ngrx/store';

const startLoading = createAction('[UI] Start loading');
const stopLoading = createAction('[UI] Stop loading');

const actions = { startLoading, stopLoading };

const actionsUnions = union(actions);

export type UIActionsType = typeof actionsUnions;
export const UIActions = actions;
