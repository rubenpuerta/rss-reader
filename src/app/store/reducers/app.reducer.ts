import { InjectionToken } from '@angular/core';
import { ActionReducerMap } from '@ngrx/store';

import { rssReducer, RssState } from '@store/reducers/rss.reducer';
import { CommonUIState, uiReducer } from '@store/reducers/ui.reducer';

export interface AppState {
	rssItems: RssState;
	commonUI: CommonUIState;
}

export const appReducer = new InjectionToken<ActionReducerMap<AppState>>('App reducers token', {
	factory: () => ({
		rssItems: rssReducer,
		commonUI: uiReducer
	})
});
