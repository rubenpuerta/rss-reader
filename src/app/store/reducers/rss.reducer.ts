import { createReducer, on } from '@ngrx/store';

import { NewsItem } from '@models/rss-news.model';
import { DataLoadActions } from '@store/actions/data-load.actions';

export interface RssState {
	news: NewsItem[];
}

export const rssInitialState: RssState = {
	news: []
};

export const rssReducer = createReducer(
	rssInitialState,
	on(DataLoadActions.initialDataLoadSuccess, (state, { news }) => ({
		...state,
		news
	}))
);
