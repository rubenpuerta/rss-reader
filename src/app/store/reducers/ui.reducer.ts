import { createReducer, on } from '@ngrx/store';

import { UIActions } from '@store/actions';

export interface CommonUIState {
	isLoading: boolean;
}

export const commonUIInitialState: CommonUIState = {
	isLoading: false
};

export const uiReducer = createReducer(
	commonUIInitialState,
	on(UIActions.startLoading, (state) => ({ ...state, isLoading: true })),
	on(UIActions.stopLoading, (state) => ({ ...state, isLoading: false }))
);
