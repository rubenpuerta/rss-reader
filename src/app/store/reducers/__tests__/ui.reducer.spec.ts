import { UIActions } from '@store/actions';
import { commonUIInitialState, CommonUIState, uiReducer } from '@store/reducers';

describe('uiReducer', () => {
	let initialState: CommonUIState;

	beforeEach(() => {
		initialState = { ...commonUIInitialState };
	});

	it('should contain initial state', () => {
		const result = uiReducer(undefined, { type: '' } as any);
		expect(result).toMatchSnapshot();
	});
	describe('UIActions.startLoading', () => {
		it('should set isLoading to true', () => {
			const result = uiReducer(initialState, UIActions.startLoading());
			expect(result).toMatchSnapshot();
		});
	});

	describe('UIActions.stopLoading', () => {
		it('should set isLoading to false', () => {
			const result = uiReducer(initialState, UIActions.stopLoading());
			expect(result).toMatchSnapshot();
		});
	});
});
