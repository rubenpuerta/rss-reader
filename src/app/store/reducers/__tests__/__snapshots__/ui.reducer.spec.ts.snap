// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiReducer UIActions.startLoading should set isLoading to true 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`uiReducer UIActions.stopLoading should set isLoading to false 1`] = `
Object {
  "isLoading": false,
}
`;

exports[`uiReducer should contain initial state 1`] = `
Object {
  "isLoading": false,
}
`;
